pkg/entrypoint/entrypointer.go

```diff
@@ -405,41 +405,43 @@ func replaceEnv(stepDir string) error {
 	for _, e := range os.Environ() {
 		pair := strings.SplitN(e, "=", 2)
 		matches := resultref.StepResultRegex.FindAllStringSubmatch(pair[1], -1)
+		v := pair[1]
 		for _, m := range matches {
 			replaceWith, _, err := findReplacement(stepDir, m[0])
 			if err != nil {
 				return err
 			}
-			os.Setenv(pair[0], strings.ReplaceAll(pair[1], m[0], replaceWith))
+			v = strings.ReplaceAll(v, m[0], replaceWith)
 		}
+		os.Setenv(pair[0], v)
 	}
 	return nil
 }
 
 // replaceCommandAndArgs performs replacements for step results in e.Command
 func replaceCommandAndArgs(command []string, stepDir string) ([]string, error) {
-	newCommand := []string{}
+	var newCommand []string
+flag:
```

Comment:
dumb question: is this the best practice to have this here?

Reply:
Wondering if we change L439 to `newC = replaceWithArray...` can we move off this flag? - but will definitely leave for your decision on the implementation.

Reply:
I changed this a little to make it more readable